puppet/zulip/files/hooks/pre-deploy.d/sentry

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/bash
+set -e
+set -u
+
+if ! grep -q 'SENTRY_DSN' /etc/zulip/settings.py; then
+    echo "sentry: No DSN configured!  Set SENTRY_DSN in /etc/zulip/settings.py"
+    exit 0
+fi
+
+if ! SENTRY_AUTH_TOKEN=$(crudini --get /etc/zulip/zulip-secrets.conf secrets sentry_release_auth_token); then
+    echo "sentry: No release auth token set!  Set sentry_release_auth_token in /etc/zulip/zulip-secrets.conf"
+    exit 0
+fi
+
+if ! SENTRY_ORG=$(crudini --get /etc/zulip/zulip.conf sentry organization); then
+    echo "sentry: No organization set!  Set sentry.organization in /etc/zulip/zulip.conf"
+    exit 0
+fi
+
+if ! SENTRY_PROJECT=$(crudini --get /etc/zulip/zulip.conf sentry project); then
+    echo "sentry: No project set!  Set setry.project in /etc/zulip/zulip.conf"
+    exit 0
+fi
+
+if ! which sentry-cli >/dev/null; then
+    echo "sentry: No sentry-cli installed!"
+    exit 0
+fi
+
+NEW_VERSION="$2"
+
+MERGE_BASE=""
+if [ "$(git rev-parse --is-inside-work-tree 2>/dev/null || true)" = "true" ]; then
+    MERGE_BASE_DESCRIBED=$(head -n2 zulip-git-version | tail -1)
+    if [[ "$MERGE_BASE_DESCRIBED" =~ ^.*-g([0-9a-f]{7,})$ ]]; then
+        MERGE_BASE=$(git rev-parse "${BASH_REMATCH[1]}")
+    else
+        MERGE_BASE=$(git rev-parse "$MERGE_BASE_DESCRIBED")
+    fi
+fi
```

Comment:
Hmm, I'm not totally sure I understand this logic's intent -- is it doing the same thing as `tools/cache-zulip-git-version`, or something different? Might be a good opportunity for a comment.

Reply:
This is taking a `git describe` and changing it back into a commit-hash.  We're reading the output of `tools/cache-zulip-git-version` (which is written to `zulip-git-version`), parsing our the mergepoint it found, and changing it into a commit hash.

I'll add a comment.